src/main/java/faang/school/projectservice/service/ProjectService.java

```diff
@@ -0,0 +1,126 @@
+package faang.school.projectservice.service;
+
+import faang.school.projectservice.dto.project.ProjectDto;
+import faang.school.projectservice.exception.DataAccessException;
+import faang.school.projectservice.jpa.ProjectJpaRepository;
+import faang.school.projectservice.mapper.ProjectMapper;
+import faang.school.projectservice.model.Project;
+import faang.school.projectservice.model.ProjectStatus;
+import faang.school.projectservice.model.ProjectVisibility;
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class ProjectService {
+
+    private final ProjectJpaRepository projectJpaRepository;
+
+    private final ProjectMapper projectMapper;
+
+
+    public ProjectDto save(ProjectDto projectDto, long requestUserId) {
+        if (projectJpaRepository.existsByOwnerIdAndName( requestUserId, projectDto.getName() )) {
+            throw new IllegalStateException( "User ID " + requestUserId + "is already a member of the team associated with the existing project." );
+        }
+        Project projectToSave = projectMapper.toEntity( projectDto );
+        projectToSave.setStatus( ProjectStatus.CREATED );
+        projectToSave.setOwnerId( requestUserId );
+        projectToSave.setVisibility( projectDto.getVisibility() == null ? ProjectVisibility.PUBLIC : projectDto.getVisibility() );
+        Project savedProject = projectJpaRepository.save( projectToSave );
+
+        return projectMapper.toDto( savedProject );
+    }
+
+    public ProjectDto updateProject(long projectId, String description, ProjectStatus status, long requestUserId) {
+        Project project = projectJpaRepository.findById( projectId )
+                .orElseThrow( () -> new EntityNotFoundException( "Project with id " + projectId + " does not exist" ) );
+
+        checkUserIsMemberOrThrowException( project, requestUserId );
+
+        project.setStatus( status != null ? status : project.getStatus() );
+        project.setDescription( description != null ? description : project.getDescription() );
+
+
+        projectJpaRepository.save( project );
+        return projectMapper.toDto( project );
+    }
+
+    public List<ProjectDto> findProjectsByName(String name, long requestUserId) {
```

Comment:
It would look much better if you use separate classes for filters, as shown on platform (section 6.1, second video)